module/reset_domain/src/mod_reset_domain.c

```diff
@@ -0,0 +1,189 @@
+/*
+ * Arm SCP/MCP Software
+ * Copyright (c) 2019-2020, Arm Limited and Contributors. All rights reserved.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Description:
+ *     Reset domain HAL
+ */
+
+#include <fwk_mm.h>
+#include <fwk_module.h>
+#include <fwk_module_idx.h>
+#include <fwk_notification.h>
+#include <mod_reset_domain.h>
+
+/*
+ * Module and devices contexts for Reset Domain
+ */
+
+/* Device context */
+struct rd_dev_ctx {
+    const struct mod_reset_domain_dev_config *config;
+    struct mod_reset_domain_drv_api *driver_api;
+};
+
+/* Module context */
+struct mod_rd_ctx {
+    const struct mod_reset_domain_config *config;
+    struct rd_dev_ctx *dev_ctx_table;
+    unsigned int dev_count;
+};
+
+/*
+ * Internal variables
+ */
+static struct mod_rd_ctx module_reset_ctx;
+
+/*
+ * API functions
+ */
+static int set_reset_state(fwk_id_t reset_dev_id,
+                           uint32_t flags,
+                           uint32_t reset_state,
+                           uintptr_t cookie)
+{
+    struct rd_dev_ctx *reset_ctx;
+    unsigned int reset_domain_idx = fwk_id_get_element_idx(reset_dev_id);
+
+    reset_ctx = &module_reset_ctx.dev_ctx_table[reset_domain_idx];
+
+    return reset_ctx->driver_api->set_reset_state(reset_ctx->config->driver_id,
+                                                  flags, reset_state, cookie);
+}
+
+/* HAL API */
+static const struct mod_reset_domain_drv_api reset_api = {
+    .set_reset_state = set_reset_state,
+};
+
+static int reset_issued_notify(fwk_id_t dev_id,
+                               uint32_t reset_state,
+                               uintptr_t cookie)
+{
+    int domain_id = -1;
+    unsigned int i;
+    struct rd_dev_ctx *reset_ctx;
+    unsigned int notification_count;
+    struct fwk_event notification_event = {
+        .id = module_reset_ctx.config->notification_id,
+        .source_id = fwk_module_id_reset_domain,
+    };
+
+    struct mod_reset_domain_notify_event_params* params =
+        (struct mod_reset_domain_notify_event_params*)
+        notification_event.params;
+
+    for (i = 0; i < module_reset_ctx.dev_count; i++) {
```

Comment:
can you please put a line of comment what this loop is doing?

Reply:
done